mypy/semanal.py

```diff
@@ -3109,6 +3109,10 @@ def leave(self) -> None:
     def is_func_scope(self) -> bool:
         return self.locals[-1] is not None
 
+    # Are we underneath a function scope, even if we are in a nested class also
+    def is_nested_func_scope(self) -> bool:
```

Comment:
I was a bit confused by the name. Maybe rename to `is_nested_within_func_scope`?